src/calendar/calendar-day.tsx

```diff
@@ -18,9 +21,16 @@ import isBetween from "dayjs/plugin/isBetween";
 dayjs.extend(isBetween);
 
 export type DayVariant = "default" | "other-month" | "today";
+type HoverDirection =
+    | "hover-start"
+    | "hover-end"
+    | "start-overlap"
+    | "end-overlap"
+    | "full-start-overlap"
+    | "full-end-overlap";
```

Comment:
nitpick, keep the naming consistent i.e. `overlap-start` -> `overlap-start` and `full-start-overlap` to `full-overlap-start`